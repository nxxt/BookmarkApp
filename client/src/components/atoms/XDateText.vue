<template>
  <div class="x-date-text d-flex align-center">
    <v-icon class="mr-1" x-small>mdi-calendar-text</v-icon>
    <span> {{ text | dateFormat }}</span>
  </div>
</template>

<script lang="ts">
import { Component, Prop, Vue } from 'vue-property-decorator'

@Component
export default class XDateText extends Vue {
  @Prop({ type: String, default: 'date' })
  text!: string
}
</script>
