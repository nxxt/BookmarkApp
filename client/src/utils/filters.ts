import IBookmark from '~~/types/IBookmark'
import ITag from '~~/types/ITag'
import IFolder from '~~/types/IFolder'

const filters: { [key: string]: Function } = {
  bookmarkTitle: (bookmark: IBookmark) => {
    return bookmark && bookmark.title ? bookmark.title : ''
  },
  tagName: (tag: ITag) => {
    return tag && tag.name ? `#${tag.name}` : ''
  },
  folderName: (folder: IFolder) => {
    return folder && folder.name ? folder.name : ''
  },
  textOverflow: (str: string) => {
    return str.length > 140 ? `${str.substring(0, 140)}â€¦` : str
  },
  dateFormat: (str: string) => {
    return str.substring(0, 10)
  }
}

export default filters
