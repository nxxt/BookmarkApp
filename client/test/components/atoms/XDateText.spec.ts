// import Vue from 'vue'
import Vuetify from 'vuetify'
import { mount, createLocalVue } from '@vue/test-utils'
import XDateText from '@/components/atoms/XDateText.vue'
import filters from '@/utils/filters'

const localVue = createLocalVue()
Object.keys(filters).forEach((key: string) =>
  localVue.filter(key, filters[key])
)

describe('XDateText.vue', () => {
  let vuetify: typeof Vuetify
  beforeEach(() => {
    vuetify = new Vuetify()
  })

  const propsData = {
    text: '2020-06-24T00:00:00'
  }

  const factory = (options?: any) => {
    return mount(XDateText, {
      localVue,
      vuetify,
      ...options
    })
  }

  it('props', () => {
    const wrapper = factory({ propsData })
    expect(wrapper.props()).toStrictEqual(propsData)
  })

  describe('template parts', () => {
    it('text', () => {
      const wrapper = factory({ propsData })
      const elm = wrapper.find('span')
      expect(elm.text()).toBe(`${filters.dateFormat(propsData.text)}`)
    })
  })

  describe('the entire template', () => {
    it('snapshot', () => {
      const wrapper = factory({ propsData })
      expect(wrapper.html()).toMatchSnapshot()
    })
  })
})
