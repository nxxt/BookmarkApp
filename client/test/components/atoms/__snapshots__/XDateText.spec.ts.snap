// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`XDateText.vue the entire template snapshot 1`] = `"<div class=\\"x-date-text d-flex align-center\\"><i aria-hidden=\\"true\\" class=\\"v-icon notranslate mr-1 mdi mdi-calendar-text theme--light\\" style=\\"font-size: 12px;\\"></i> <span> 2020-06-24</span></div>"`;
